packages/typography/src/views/lists/List.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { createContext } from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import { isRtl, retrieveTheme } from '@zendeskgarden/react-theming';
+
+const COMPONENT_ID = 'typography.list';
+
+const TYPE_ORDERED = {
+  DECIMAL: 'decimal',
+  DECIMAL_LEADING_ZERO: 'decimal-leading-zero',
+  LOWER_ALPHA: 'lower-alpha',
+  LOWER_ROMAN: 'lower-roman',
+  UPPER_ALPHA: 'upper-alpha',
+  UPPER_ROMAN: 'upper-roman'
+};
+
+const TYPE_UNORDERED = {
+  CIRCLE: 'circle',
+  DISC: 'disc',
+  SQUARE: 'square'
+};
+
+const SIZE = {
+  SMALL: 'small',
+  MEDIUM: 'medium',
+  LARGE: 'large'
+};
+
+const listAttributes = {
+  'data-garden-id': COMPONENT_ID,
+  'data-garden-version': PACKAGE_VERSION
+};
+
+const listCSS = css`
+  direction: ${props => (isRtl(props) ? 'rtl' : 'ltr')};
+  margin: 0;
+  /* stylelint-disable-next-line property-no-unknown */
+  margin-${props => (isRtl(props) ? 'right' : 'left')}: 24px;
+  padding: 0;
+  list-style-position: outside;
+
+  ${props => retrieveTheme(COMPONENT_ID, props)};
+`;
+
+export const StyledOrderedList = styled.ol.attrs(listAttributes)`
+  ${listCSS};
+  list-style-type: ${props => Object.values(TYPE_ORDERED).indexOf(props.type) !== -1 && props.type};
```

Comment:
This seems kind of expensive to happen every render. This logic could live in the component and not at the `styled-components` level.